api/v3/utils.php

```diff
@@ -1931,17 +1998,23 @@ function _civicrm_api3_api_match_pseudoconstant(&$params, $entity, $fieldName, $
     }
     $options = civicrm_api($entity, 'getoptions', array('version' => 3, 'field' => $fieldInfo['name'], 'context' => 'validate'));
     $options = CRM_Utils_Array::value('values', $options, array());
+
+    if (count($options) == 0 && isset($pseudoconstant['table'])) {
+        $pseudoParams = $pseudoconstant;
+        unset($pseudoParams['table']);
+        $options = CRM_Core_PseudoConstant::get(CRM_Core_DAO_AllCoreTables::getClassForTable($pseudoconstant['table']), $fieldName, $pseudoParams);
```

Comment:
Those attribute(s) which has pseudo property as 'table' and specific keyColumn/LabelColumn value didn't fall under 'getoptions' api. Like in xml you will find that I have added pseudoconstant for contrubution_soft.pcp_id
<pseudoconstant>
+      <table>civicrm_pcp</table>
+      <keyColumn>id</keyColumn>
+      <labelColumn>title</labelColumn>
+    </pseudoconstant>
So no options (id=>title) get retrieved via getoptions.api so I handled it in here to support 'pseudoconstant' property too not ONLY 'option'

  I'll surely add a comment above this code.